api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java

```diff
@@ -66,6 +67,70 @@ public T getPayload() {
         };
     }
 
+    /**
+     * Create a message with the given payload, ack and nack functions.
+     *
+     * @param payload The payload.
+     * @param ack The ack function, this will be invoked when the returned messages {@link #ack()} method is invoked.
+     * @param nack The negative-ack function, this will be invoked when the returned messages {@link #nack(Throwable)}
+     *        method is invoked.
+     * @param <T> the type of payload
+     * @return A message with the given payload, metadata, ack and nack functions.
+     */
+    static <T> Message<T> of(T payload,
+                             Supplier<CompletionStage<Void>> ack, Function<Throwable, CompletionStage<Void>> nack) {
+        return new Message<T>() {
+            @Override
+            public T getPayload() {
+                return payload;
+            }
+
+            @Override
+            public Supplier<CompletionStage<Void>> getAck() {
+                return ack;
+            }
+
+            @Override
+            public Function<Throwable, CompletionStage<Void>> getNack() {
+                return nack;
+            }
+        };
+    }
+
+    /**
+     * Creates a new instance of {@link Message} with the specified payload.
+     * The metadata and ack/nack functions are taken from the current {@link Message}.
+     *
+     * @param payload the new payload.
+     * @param <P> the type of the new payload
+     * @return the new instance of {@link Message}
+     */
+    default <P> Message<P> withPayload(P payload) {
+        return Message.of(payload, getAck(), getNack());
+    }
+
+    /**
+     * Creates a new instance of {@link Message} with the given acknowledgement supplier.
+     * The payload, metadata and nack function are taken from the current {@link Message}.
+     *
+     * @param supplier the acknowledgement supplier
+     * @return the new instance of {@link Message}
+     */
+    default Message<T> withAck(Supplier<CompletionStage<Void>> supplier) {
```

Comment:
To be consistent with the following method, should `supplier` be updated to `ack`, to increase the readability?

Reply:
Do you mean the name of the parameter? If so, yes, not opinion.